storage2/src/snapshot.rs

```diff
@@ -130,18 +138,20 @@ impl StateRead for Snapshot {
 
 /// A reader interface for rocksdb. NOTE: it is up to the caller to ensure consistency between the
 /// rocksdb::DB handle and any write batches that may be applied through the writer interface.
-impl TreeReader for Inner {
+impl TreeReader for Snapshot {
```

Comment:
The `TreeReader` implementation is now on the `Snapshot`, since the `Inner` type is hidden from the rest of the crate.  But we don't expose it outside of the crate anyways.